### PR TITLE
[feat](Nereids): lazy get expression map when comparing hypergraph

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

lazy get ExpressionMap when comparing hypergraph

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/LogicalCompatibilityContext.java [hunk 1]
```diff
@@ -92,31 +92,28 @@ public BiMap<Integer, Integer> getQueryToViewNodeIDMapping() {
...
-        if (queryToViewAllExpressionMapping != null) {
-            return queryToViewAllExpressionMapping;
-        }
-        queryToViewAllExpressionMapping = HashBiMap.create();
-        queryToViewAllExpressionMapping.putAll(getQueryToViewJoinEdgeExpressionMapping());
-        queryToViewAllExpressionMapping.putAll(getQueryToViewNodeExpressionMapping());
-        queryToViewAllExpressionMapping.putAll(getQueryToViewFilterEdgeExpressionMapping());
-        return queryToViewAllExpressionMapping;
+    public Expression getQueryJoinExprFromView(Expression viewJoinExpr) {
+        return queryToViewJoinEdgeExpressionMappingSupplier.get().inverse().get(viewJoinExpr);
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/LogicalCompatibilityContext.java:96]: `inverse()` need calc, Maybe we should construct `viewToQueryJoinEdgeExpressionMapping`  field in context for performance

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/LogicalCompatibilityContext.java [hunk 2]
```diff
@@ -92,31 +92,28 @@ public BiMap<Integer, Integer> getQueryToViewNodeIDMapping() {
...
+    }
+
+    public Expression getViewJoinExprFromQuery(Expression queryJoinExpr) {
+        return queryToViewJoinEdgeExpressionMappingSupplier.get().get(queryJoinExpr);
+    }
+
+    public Expression getQueryFilterExprFromView(Expression viewJoinExpr) {
+        return queryToViewFilterEdgeExpressionMappingSupplier.get().inverse().get(viewJoinExpr);
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/LogicalCompatibilityContext.java:104]: the same as above

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/LogicalCompatibilityContext.java [hunk 3]
```diff
@@ -92,31 +92,28 @@ public BiMap<Integer, Integer> getQueryToViewNodeIDMapping() {
...
-    public BiMap<Expression, Expression> getQueryToViewJoinEdgeExpressionMapping() {
-        return queryToViewJoinEdgeExpressionMappingSupplier.get();
+    public Expression getViewFilterExprFromQuery(Expression queryJoinExpr) {
+        return queryToViewFilterEdgeExpressionMappingSupplier.get().get(queryJoinExpr);
     }
 
-    public BiMap<Expression, Expression> getQueryToViewNodeExpressionMapping() {
-        return queryToViewNodeExpressionMappingSupplier.get();
+    public Expression getQueryNodeExpFromView(Expression viewJoinExpr) {
+        return queryToViewNodeExpressionMappingSupplier.get().inverse().get(viewJoinExpr);
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/LogicalCompatibilityContext.java:112]: the same as above

**Comment by XieJiann** [comment]
run buildall

**Review by seawinde — APPROVED** [review]

**Comment by morrySnow** [comment]
run performance

**Review by morrySnow — APPROVED** [review]

**Pull request merged**